persistent/Database/Persist/Sql/Run.hs

```diff
@@ -28,15 +86,16 @@ import Database.Persist.Sql.Raw
 runSqlPool
     :: (MonadUnliftIO m, BackendCompatible SqlBackend backend)
     => ReaderT backend m a -> Pool backend -> m a
-runSqlPool r pconn = withRunInIO $ \run -> withResource pconn $ run . runSqlConn r
+runSqlPool r pconn = with (acquireSqlConnFromPool pconn) $ runReaderT r
```

Comment:
The change from `runSqlConn` to `runReaderT` drops the logic around transactions and exception handling. Why'd this get changed?

```suggestion
runSqlPool r pconn = with (acquireSqlConnFromPool pconn) $ runSqlConn r
```